### PR TITLE
leave out members that have left the cluster : RaftGroupMembershipMan…

### PR DESCRIPTION
**Pull request by fbarotov**
Fixes https://github.com/hazelcast/hazelcast/issues/19492

Co-Authored-By: Jaromir Hamala <jaromir.hamala@gmail.com>

### CODE REVIEW

**Comment by jerrinot** [comment]
@fbarotov the fix is OK. have you thought how to test it?

**Comment by fbarotov** [comment]
@jerrinot thanks for the review. Yes, I extended CPGroupRebalanceTest to also test the scenario where a member is terminated and the leadership rebalance process should succeed.

File: hazelcast/src/test/java/com/hazelcast/cp/internal/CPGroupRebalanceTest.java [hunk 1]
```diff
@@ -86,6 +90,43 @@ public void test() throws Exception {
...
+            for (CPGroupId groupId : groupIds) {
+                waitAllForLeaderElection(aliveInstances, groupId);
+            }
+
+            rebalanceLeaderships(newMetadataLeader);
+
+            Map<CPMember, Collection<CPGroupId>> leadershipsMap = getLeadershipsMap(newMetadataLeader, aliveCpMembers);
+            for (Entry<CPMember, Collection<CPGroupId>> entry : leadershipsMap.entrySet()) {
+                int count = entry.getValue().size();
+                assertEquals(leadershipsString(leadershipsMap), 25, count);
```
Comment (sancar) [inline hazelcast/src/test/java/com/hazelcast/cp/internal/CPGroupRebalanceTest.java:127]:
```suggestion
                assertEquals(leadershipsString(leadershipsMap), (groupCount - 1) / groupSize , count);
```
One minor comment to help the reader to understand where `25` come from.

**Review by sancar — APPROVED** [review]

**Comment by vbekiaris** [comment]
@jerrinot can you complete your review?

**Review by jerrinot — APPROVED** [review]

**Review by jerrinot — APPROVED** [review]

**Comment by fbarotov** [comment]
We discussed it with Vassilis and Jaromir, let's merge this PR after running Jepsen tests

**Comment by Holmistr** [comment]
@olukas Would it be possible to execute the Jepsen tests on this branch?

**Comment by olukas** [comment]
@Holmistr in theory - yes. We can adjust any of our Jepsen jobs to run from the certain branch etc:
- http://jenkins.hazelcast.com/job/jepsen-single-test/
- http://jenkins.hazelcast.com/job/jepsen

The problem is we are able to execute Jepsen tests but they are not too stable. See https://hazelcast.atlassian.net/browse/HZ-746?focusedCommentId=59531 for more details.

**Pull request closed without merging**